DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToHost/Raw/Geometry/PolylineConverterToHost.cs

```diff
@@ -0,0 +1,96 @@
+
+public class PolylineConverterToHost : ITypedConverter<SOG.Polyline, DB.CurveArray>
+{
+  private readonly ITypedConverter<SOG.Line, DB.Line> _lineConverter;
+  private readonly ScalingServiceToHost _scalingService;
+  private readonly IRevitConversionContextStack _contextStack;
+
+  public PolylineConverterToHost(
+    ITypedConverter<SOG.Line, DB.Line> lineConverter,
+    ScalingServiceToHost scalingService,
+    IRevitConversionContextStack contextStack
+  )
+  {
+    _lineConverter = lineConverter;
+    _scalingService = scalingService;
+    _contextStack = contextStack;
+  }
+
+  public CurveArray Convert(Polyline target)
+  {
+    var curveArray = new CurveArray();
+    if (target.value.Count == 6)
+    {
+      // Polyline is actually a single line
+      TryAppendLineSafely(curveArray, new SOG.Line(target.value, target.units));
+    }
+    else
+    {
+      var pts = target.GetPoints();
+      var lastPt = pts[0];
+      for (var i = 1; i < pts.Count; i++)
+      {
+        var success = TryAppendLineSafely(curveArray, new SOG.Line(lastPt, pts[i], target.units));
+        if (success)
+        {
+          lastPt = pts[i];
+        }
+      }
+
+      if (target.closed)
+      {
+        TryAppendLineSafely(curveArray, new SOG.Line(pts[^1], pts[0], target.units));
+      }
+    }
+    return curveArray;
+  }
+
+  /// <summary>
+  /// Checks if a Speckle <see cref="SOG.Line"/> is too sort to be created in Revit.
+  /// </summary>
+  /// <remarks>
+  /// The length of the line will be computed on the spot to ensure it is accurate.
+  /// </remarks>
+  /// <param name="line">The <see cref="SOG.Line"/> to be tested.</param>
+  /// <returns>true if the line is too short, false otherwise.</returns>
+  public bool IsLineTooShort(SOG.Line line)
+  {
+    var scaleToNative = _scalingService.ScaleToNative(SOG.Point.Distance(line.start, line.end), line.units);
+    return scaleToNative < _contextStack.Current.Document.Application.ShortCurveTolerance;
+  }
+
+  /// <summary>
+  /// Attempts to append a Speckle <see cref="SOG.Line"/> onto a Revit <see cref="CurveArray"/>.
+  /// This method ensures the line is long enough to be supported.
+  /// It will also convert the line to Revit before appending it to the <see cref="CurveArray"/>.
+  /// </summary>
+  /// <param name="curveArray">The revit <see cref="CurveArray"/> to add the line to.</param>
+  /// <param name="line">The <see cref="SOG.Line"/> to be added.</param>
+  /// <returns>True if the line was added, false otherwise.</returns>
+  public bool TryAppendLineSafely(CurveArray curveArray, SOG.Line line)
+  {
+    if (IsLineTooShort(line))
+    {
+      // poc : logging "Some lines in the CurveArray where ignored due to being smaller than the allowed curve length."
+      return false;
+    }
+    try
+    {
+      curveArray.Append(_lineConverter.Convert(line));
+      return true;
+    }
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      // poc : logging
```

Comment:
@JR-Morgan not sure we could get anything meaningful into logs or the report from here??

Reply:
I would like us to discuss if we think a "partial conversion" with missing curves is a good UX here.

If we agree that there is a need for it, @didimitrie and I have already discussed how we would bring a "succeeded with warnings" system to converters, and this is where logging should exist, not directly in converters.

However I strongly object to having any indescriminate `catch(Exception ex) when (!ex.IsFatal())` involved inside specific conversions. We should decide the expected cases that we're expecting, and let unexpected ones fail the entire object. It gets very messy if we are were to allow unpredictable failures at many levels of our application

Reply:
If @didimitrie and @BovineOx agree, I'm not sure we're ready yet do "succeeded with warnings", so for the sake of this pr, I would suggest we err on the side of failing the entire object, then we can always being a more granular conversion when we have the appropriate tooling for informing the user

Reply:
okay, try catch removed 👍